hs/WebParsing/PrerequisiteParsing.hs

```diff
@@ -23,53 +23,53 @@ import qualified Data.Text as T
 -- returns (before, delim, after). or (input, "","") if no match occurs
 matchDelim :: String -> (String, String, String)
 matchDelim prereqs =
-  let pat = "[;,]" :: String
-  in prereqs =~ pat
+    let pat = "[;,]" :: String
+    in prereqs =~ pat
 
 -- | returns true if the string begins inside a parenthesized expression.
 -- e.g  isntDelim "CSC458)" == True
 --      isntDelim "(STA247, STA248)" == False
 isntDelim :: String -> Bool
 isntDelim rest =
-  let pat = "^.*\\)" :: String
-  in rest =~ pat
+    let pat = "^.*\\)" :: String
+    in rest =~ pat
 
 -- |Splits a PrereqString by delimeters ';'' ','.
 toPreExprs :: String -> String -> [String]
 toPreExprs str expr  =
-  let (beforeStr, delimStr, afterStr) = matchDelim str
-  in case (beforeStr, delimStr, afterStr) of
-    ("","","") -> [] --if (expr == "") then [] else [expr]
-    (before, "", "") -> [before++expr]
-    (before, ",", after) -> if (isntDelim after)
-                            then toPreExprs after (expr ++ before)
-                            else (expr++before):(toPreExprs after "")
-    (before, ";", after) -> (expr++before):(toPreExprs after "")
+    let (beforeStr, delimStr, afterStr) = matchDelim str
+    in case (beforeStr, delimStr, afterStr) of
+          ("","","") -> [] --if (expr == "") then [] else [expr]
+          (before, "", "") -> [before++expr]
+          (before, ",", after) -> if (isntDelim after)
+                                  then toPreExprs after (expr ++ before)
+                                  else (expr++before):(toPreExprs after "")
+          (before, ";", after) -> (expr++before):(toPreExprs after "")
 
 -- | attempts to match a course in given string. returns (before, course, after)
 -- if no match occurs (input, "", "")
 matchCourse :: String -> (String, String, String)
 matchCourse prereqs =
-  let pat = "[A-Z]{3}[0-9]{3}[HY][0-9]" :: String
-  in prereqs =~ pat
+    let pat = "[A-Z]{3}[0-9]{3}[HY][0-9]" :: String
+    in prereqs =~ pat
 
 -- | converts a string representing a prerequisite expression into a prerequisite
 -- expression. Extracts all course names found within the string, and returns them
 -- in a string.
 toPrereq :: String -> T.Text
 toPrereq expr =
-  let (beforeExpr, courseExpr, afterExpr) = matchCourse expr
-  in case (beforeExpr, courseExpr, afterExpr) of
-     (_, "", "") -> ""
-     --guaranteed match
-     (_, course, "") -> T.pack course
-     (_, course, after) ->  T.concat [(T.pack course), " ", (toPrereq after)]
+    let (beforeExpr, courseExpr, afterExpr) = matchCourse expr
+    in case (beforeExpr, courseExpr, afterExpr) of
```

Comment:
The 3-space indentation isn't great. I think it makes more sense to start multi-line `in` blocks on a new line, indented 4 spaces.